src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java

```diff
@@ -276,16 +278,17 @@ public CcCompilationContext getCcCompilationContext() {
 
   /** Creates a CcCompilationHelper that outputs artifacts in a given configuration. */
   public CcCompilationHelper(
-      ActionRegistry actionRegistry,
-      ActionConstructionContext actionConstructionContext,
-      Label label,
-      @Nullable Artifact grepIncludes,
-      CppSemantics semantics,
-      FeatureConfiguration featureConfiguration,
-      SourceCategory sourceCategory,
-      CcToolchainProvider ccToolchain,
-      FdoContext fdoContext,
-      BuildConfiguration buildConfiguration) {
+          ActionRegistry actionRegistry,
+          ActionConstructionContext actionConstructionContext,
+          Label label,
+          @Nullable Artifact grepIncludes,
+          CppSemantics semantics,
+          FeatureConfiguration featureConfiguration,
+          SourceCategory sourceCategory,
+          CcToolchainProvider ccToolchain,
+          FdoContext fdoContext,
+          BuildConfiguration buildConfiguration,
+          @Nullable RuleContext ruleContext) {
```

Comment:
Why nullable?

Reply:
Because it can be null in case of sandwich, like we understood it with Jakob yesterday. See: https://github.com/bazelbuild/bazel/pull/9267/files/49ca1c4f3364a34a398a03b39b29a83552ce1e69#r318975035

I'll also have a look at ActionConstructionContext, maybe we could use that => this @Nullable will not be needed at all, as well as RuleContext